demonstrations/tutorial_qubitization.py

```diff
@@ -0,0 +1,171 @@
+
+Block Encoding
+----------------
+
+First, we introduce some useful concepts about block encodings.
+Given a Hamiltonian :math:`\mathcal{H}`, we define as a block encoding any operator that embeds :math:`\mathcal{H}`
+inside the matrix associated to the circuit (up to a normalization factor):
+
+.. math::
+    \text{Block Encoding}_\mathcal{H} \rightarrow \begin{bmatrix} \mathcal{H} / \lambda & \cdot \\ \cdot & \cdot \end{bmatrix}.
+
+Given an eigenvector of :math:`\mathcal{H}` such that :math:`\mathcal{H}| \phi \rangle = E | \phi \rangle`, we deduce that any :math:`\text{Block Encoding}_\mathcal{H}`
+generates a state:
+
+.. math::
+    |\Psi\rangle = \frac{E}{\lambda}|0\rangle |\phi\rangle + \sqrt{1 - \left( \frac{E}{\lambda}\right)^2} |\phi^{\perp}\rangle,
+
+where :math:`|\phi^{\perp}\rangle` is a state orthogonal to :math:`|0\rangle |\phi\rangle`,
+and :math:`E` is the eigenvalue. The advantage of expressing :math:`|\Psi\rangle` as the sum of two orthogonal states is that it can be represented
+in a two-dimensional space  — an idea that we have explored in our `Amplitude Amplification <https://pennylane.ai/qml/demos/tutorial_intro_amplitude_amplification/>`_ demo. The state :math:`|\Psi\rangle` forms an angle :math:`\theta =\arccos {\frac{E}{\lambda}}` with respect to the axis defined by the initial state :math:`|0\rangle |\phi\rangle`, as shown in the image below.
+
+.. figure:: ../_static/demonstration_assets/qubitization/qubitization0.jpeg
+    :align: center
+    :width: 50%
+    :target: javascript:void(0)
+
+Qubitization as a rotation
+---------------------------
+
+Any block encoding operator manages to transform the state :math:`|0\rangle |\phi\rangle` into the state :math:`|\Psi\rangle`.
+The Qubitization operator does this by applying a rotation in that two-dimensional subspace by an angle of :math:`\theta=\arccos {\frac{E}{\lambda}}`.
+The advantage of a rotation operator is that the angle :math:`\theta` appears directly in its eigenvalues as the phases  :math:`e^{\pm i\theta}`. Therefore, if the rotation angle encodes useful information, it can be retrieved by estimating the phase of the rotation operator, for example using QPE.
+
+We now show that the Qubitization operator is a rotation by using the same idea of Amplitude Amplification:
+two reflections are equivalent to one rotation. These reflections correspond to :math:`(2|0\rangle\langle 0| - I)` and :math:`\text{PSP}_{\mathcal{H}}`, which together define our Qubitization operator.
+
+.. note::
+
+    To verify that an operator is indeed a reflection, one must square the operator and check if the result is the identity operator.
```

Comment:
```suggestion
    An operator :math:`U` is a reflection if :math:`U^2=I`.
```
I'm not a fan of thes enotes; they break the flow of reading the demo. I suggest just adding this as a sentence, e.g., recall that an operator is a reflection if...